### PR TITLE
Introduce PoseGraphInterface::GetAllSubmapPoses()

### PR DESCRIPTION
**Pull request by cschuet**

### CODE REVIEW

File: cartographer/mapping_2d/pose_graph.h [hunk 1]
```diff
@@ -102,8 +102,10 @@ class PoseGraph : public mapping::PoseGraph {
   std::vector<std::vector<int>> GetConnectedTrajectories() override;
   mapping::PoseGraph::SubmapData GetSubmapData(
       const mapping::SubmapId& submap_id) EXCLUDES(mutex_) override;
-  mapping::MapById<mapping::SubmapId, mapping::PoseGraph::SubmapData>
+  mapping::MapById<mapping::SubmapId, mapping::PoseGraphInterface::SubmapData>
   GetAllSubmapData() EXCLUDES(mutex_) override;
+  mapping::MapById<mapping::SubmapId, SubmapPose> GetAllSubmapPoses()
```
Comment (gaschler) [inline cartographer/mapping_2d/pose_graph.h]: consistency: mapping::PoseGraphInterface::SubmapPose, also in _3d/pose_graph.h

File: cartographer/mapping_2d/pose_graph.cc [hunk 2]
```diff
@@ -642,17 +642,31 @@ mapping::PoseGraph::SubmapData PoseGraph::GetSubmapData(
...
     submaps.Insert(submap_id_data.id,
                    GetSubmapDataUnderLock(submap_id_data.id));
   }
   return submaps;
 }
 
+mapping::MapById<mapping::SubmapId, mapping::PoseGraphInterface::SubmapPose>
+PoseGraph::GetAllSubmapPoses() {
+  common::MutexLocker locker(&mutex_);
+  mapping::MapById<mapping::SubmapId, SubmapPose> submaps;
```
Comment (gaschler) [inline cartographer/mapping_2d/pose_graph.cc]: naming: submap_poses? (also the other one)

File: cartographer/mapping_2d/pose_graph.h (hunk 1 above)
Comment (cschuet) [reply cartographer/mapping_2d/pose_graph.h]: As discussed SubmapData is already defined in the implementing class that's we need to add the explicit namespace to disambiguate. Here this is not necessary. And we usually don't specify it when not necessary.

**Comment by cschuet** [comment]
@wally-the-cartographer merge

**Comment by wally-the-cartographer** [comment]
Merge requested by authorized user cschuet. Merge queue now has a length of 1.

**Pull request merged**